gapic-generator-java/src/test/java/com/google/api/generator/gapic/protoparser/ParserTest.java

```diff
@@ -705,6 +710,99 @@ void parseServiceWithNoMethodsTest() {
     assertEquals("EchoWithMethods", services.get(0).overriddenName());
   }
 
+  @Test
+  void selectiveGenerationTest_shouldExcludeUnusedResourceNames() {
```

Comment:
This test verifies that unused resource names (associated with methods not in inclusion list) will not be populated to GapicContext.